components/omega/doc/design/VertCoord.md

```diff
@@ -0,0 +1,294 @@
+(omega-design-vert-coord)=
+# Vertical Coordinate
+
+## 1 Overview
+The vertical coordinate module will be responsible for computing and storing information related to the vertical mesh in Omega.
+Since Omega will be a non-Boussinesq model, the vertical independent variable will be pressure- as opposed to $z$-based.
+Similar to MPAS-Ocean's support for tilted height ($z^\star$) coordinates, Omega V1 will support a general ($p^\star$) vertical coordinate, where mass thickness $h$ varies in proportion to the total mass thickness of the water column. 
+The prognostic variable in the layered continuity equation is mass thickness, which can be used to calculate the total pressure at a given vertical level.
+The vertical height of a given level is still necessary in the model to compute sea level, the geopotential, and derivatives with respect to $z$ (vertical mixing, etc).
+The height of a layer ($\Delta z$) can be found using the mass thickness, density and known bottom elevation relative to the geoid (positive up).
+The vertical coordinate module will also serve as a container for information related to the variable number of vertical levels for a given ocean column due to variations in bottom elevation and ice shelf cavities.
+
+
+## 2 Requirements
+
+### 2.1 Requirement: Compute and store the total pressure at a given level
+
+The total pressure at the bottom of each level will be computed and stored within the vertical coordinate module.
+This involves a surface-down summation of the mass thickness variable times the gravitational acceleration, $g$, starting with the surface pressure.
+The pressure variable will be needed in the computation of the TEOS-10 equation of state and the pressure gradient.
+The variable will be registered with the `IOStreams` framework to enable it to be output during runtime.
+
+### 2.2 Requirement: Compute and store the $z$ location of a given level
+
+The $z$ location at the top of each level will be computed and stored within the vertical coordinate module.
+In the model, $z$ is defined as being positive up from the geoid.
+The $z$ location is calculated in a bottom-up summation of the mass thickness times the specific volume, starting with the known bottom elevation.
+The $z$ location of the interfaces is needed to compute the geopotential gradient and the sea level.
+The variable will be registered with the `IOStreams` framework to enable it to be output during runtime.
+
+### 2.3 Requirement: Compute and store the vertical level bounds for each horizontal cell
+
+All tendency terms require a variable vertical loop range to account for variation in bottom elevation and ice shelf cavities.
+These bounds will be computed in the vertical coordinate module and used for setting masking arrays that will be used to determine where calculations are valid in the vertical.
```

Comment:
From our last meeting, sounds like this will be loop bounds instead of masking?

Reply:
yes, I'll update that.